File: app/src/main/java/com/ababilx/routinescrapper/domain/model/RoutineDay.kt
package com.ababilx.routinescrapper.domain.model

enum class RoutineDay(val shortLabel: String, val fullLabel: String) {
    SATURDAY("Sat", "Saturday"),
    SUNDAY("Sun", "Sunday"),
    MONDAY("Mon", "Monday"),
    TUESDAY("Tue", "Tuesday"),
    WEDNESDAY("Wed", "Wednesday"),
    THURSDAY("Thu", "Thursday");

    companion object {
        fun fromName(raw: String): RoutineDay =
            entries.firstOrNull { it.name == raw.uppercase() } ?: SATURDAY
    }
}

File: lib/domain/model/routine_day.dart
enum RoutineDay {
  saturday('Sat', 'Saturday'),
  sunday('Sun', 'Sunday'),
  monday('Mon', 'Monday'),
  tuesday('Tue', 'Tuesday'),
  wednesday('Wed', 'Wednesday'),
  thursday('Thu', 'Thursday');

  const RoutineDay(this.shortLabel, this.fullLabel);

  final String shortLabel;
  final String fullLabel;

  String get wireName => name.toUpperCase();

  static RoutineDay fromName(String raw) {
    final target = raw.toUpperCase();
    for (final day in RoutineDay.values) {
      if (day.wireName == target) return day;
    }
    return RoutineDay.saturday;
  }
}
